### DISCUSSION TITLE
LinqToDb CLI add comment above Namespace

### DISCUSSION THREAD

**Discussion by croban** [linq2db/linq2db#3875] [Q&A] [answered]
How can I add a `// ReSharper disable once CheckNamespace` above namespace?

Something like: 
```
entityModel.Namespace.AddComment("// ReSharper disable once CheckNamespace")
```

**Comment by MaceWindu** [comment, marked as answer]
Such functionality not supported, but there is default comment, which could be replaced/updated using `header` option

**Reply by croban** [reply]
Unfortunately, specifying custom header has impact only on text inside `<auto-generated>` Xml Node e.g.

**cli.json**
``` 
...
"code": {
    "nrt": true,
    "namespace": "SomeCustom.Models",
    "autogenerated": true,
    "header": "// ReSharper disable once CheckNamespace"
  }
```

**produce**:
```
// ---------------------------------------------------------------------------------------------------
// <auto-generated>
// // ReSharper disable once CheckNamespace
// </auto-generated>
// ---------------------------------------------------------------------------------------------------`
````

What I saw is that Resharper as soon as `<auto-generated>` is inside code, `CheckNamespace` is turned off. 
So I will leave default header inside. 

There is additional side effect if `"autogenerated": true`:  
Suddenly I have also `#nullable enable` inside file. 
Before that nullable properties were generated but without `#nullable enable` on the top within generated file.

**Reply by MaceWindu** [reply]
Ah, true. It's only allow text inside auto-generated tag

regarding `#nullable enable`. It is generated because `auto-generated` tag disables nullable context and we need to re-enable it
